.github/workflows/release-packages.yml

```diff
@@ -1,37 +1,49 @@
-name: Release Package
+name: Release Packages
 
 on:
   push:
     branches:
+      - main
+    tags:
       - "**"
-    tags-ignore: ["**"]
 
 jobs:
-  versionning:
-    name: Versionning
+  create-version:
+    name: Create version
     runs-on: ubuntu-latest
     outputs:
-      version: ${{ steps.genver.outputs.version }}
+      version: ${{ steps.version.outputs.version }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           ref: ${{ github.ref }}
           fetch-depth: 0
 
-      - name: Generate Version
-        id: genver
+      - name: Generate version from git
+        if: ${{ !contains(github.ref, 'refs/tags') }}
+        run: |
+          VERSION=$(docker run --rm -v $(pwd):/repo codacy/git-version /bin/git-version --folder=/repo --release-branch=main)
+          echo "version=$VERSION" >> $GITHUB_ENV
+
+      - name: Get version from tag
+        if: ${{ contains(github.ref, 'refs/tags') }}
```

Comment:
How do you ensure that the tag is coherent ?
This should work only on X.Y.Z tags. Moreover, how do you ensure that the tag given is really the next tag ?
Things I do not want to see:
0.6.6 -> 0.8.0
0.6.6 -> 0.6.8

Reply:
I can't. Maintainers are responsible for publishing the correct tag (and you can do this using a PR with a review). Using an automated system is the same, you have the responsibility to add `feature` or `breaking` in a comment from one or more commit. So, if another maintainer forgot to add `breaking` in the commit message, the tag will not be correct and the workflow can be achieved by only one person. With a tag release from a PR, you must have a pair validation which I think is better.